src/Extension/CoreExtension.php

```diff
@@ -1706,7 +1717,7 @@ public static function getAttribute(Environment $env, Source $source, $object, $
                     $env->getExtension(SandboxExtension::class)->checkPropertyAllowed($object, $item, $lineno, $source);
                 }
 
-                return isset($object->$item) ? $object->$item : ((array) $object)[(string) $item];
+                return $propertyReflector ? $propertyReflector->getValue($object) : $object->$item;
```

Comment:
how will this behave for hooks ?

Reply:
this will call them, as it should for twig IMHO